File: dashboard/types/camera.ts
export interface Camera {
  id: string;
  name: string;
  location: string | null;
  rtsp_url: string;
  status: string;
  enabled: boolean;
  fps: number | null;
  resolution: string | null;
  last_seen: string | null;
  created_at: string;
  updated_at: string;
}

export interface CreateCameraRequest {
  name: string;
  location?: string;
  rtsp_url: string;
  fps?: number;
  resolution?: string;
}

export interface UpdateCameraRequest {
  name?: string;
  location?: string | null;
  rtsp_url?: string;
  fps?: number | null;
  resolution?: string | null;
  enabled?: boolean;
}

export interface CameraFilters {
  status?: string;
  search?: string;
  page?: number;
  per_page?: number;
}

File: mobile-app/lib/features/cameras/domain/entities/camera.dart
class Camera {
  final String id;
  final String name;
  final String? location;
  final String status;
  final bool enabled;
  final String rtspUrl;
  final int? fps;
  final String? resolution;
  final DateTime? lastSeen;
  final DateTime createdAt;
  final DateTime? updatedAt;

  const Camera({
    required this.id,
    required this.name,
    this.location,
    required this.status,
    required this.enabled,
    required this.rtspUrl,
    this.fps,
    this.resolution,
    this.lastSeen,
    required this.createdAt,
    this.updatedAt,
  });

  bool get isOnline => status == 'online';
}
